bin/utils.js

```diff
@@ -0,0 +1,104 @@
+const dayjs = require("dayjs");
+const { serializeError } = require("serialize-error");
+const ms = require("ms");
+const _ = require("lodash");
+const {ObjectId} = require("mongodb");
+const relativeTime = require('dayjs/plugin/relativeTime')
+const duration = require('dayjs/plugin/duration')
+
+dayjs.extend(duration)
+dayjs.extend(relativeTime)
+
+const attachExitHandlers = (exitCallback = (code = 1) => { process.exit(code)}) => {
+  // https://blog.heroku.com/best-practices-nodejs-errors
+  // https://stackoverflow.com/a/14032965/1327178
+  [`SIGINT`, `SIGTERM`, `SIGUSR1`, `SIGUSR2`].forEach((event) => {
+    process.on(event, () => {
+      console.log(`\n${event} received`);
+      exitCallback(0);
+    });
+  });
+  process
+    /* NodeJS process will exit when:
+     * - the event loop's queue is empty.
+     * - no background/asynchronous tasks remain that are capable of adding to the queue. */
+    .on('exit', () => {
+      console.log('NodeJS has nothing more to execute ...');
+      exitCallback();
```

Comment:
https://nodejs.org/api/process.html#process_event_exit

From node.js documentation 

> Listener functions must only perform synchronous operations. The Node.js process will exit immediately after calling the 'exit' event listeners causing any additional work still queued in the event loop to be abandoned.

Reply:
I think that you want this https://nodejs.org/api/process.html#event-beforeexit